include/rx/ranges.hpp

```diff
@@ -1807,6 +1807,33 @@ struct none_of {
 };
 template <class P>
 none_of(P &&)->none_of<P>;
+    
+template <class P>
+struct contains {
+    const P& pred;
+
+    constexpr explicit contains(const P& pred)
```

Comment:
you should take it by value and move it into the member. Or use the same pattern as `std::optional` and other:
```cpp
template<typename...Args>
explicit constexpr contains(std::in_place_t, Args&&... args) : pred(std::forward<Args>(args)...) {}